helm/templates/deployment.yaml

```diff
@@ -125,31 +125,43 @@ spec:
       {{ end }}
       {{ if $.Values.minio.metricsLogger.enable }}
       - name: last-action-logger
-        image: {{ $.Values.minio.metricsLogger.image }}:{{ $.Values.minio.tag | default $.Chart.AppVersion }}
+        image: {{ $.Values.minio.metricsLogger.image }}
         imagePullPolicy: {{ $.Values.minio.metricsLogger.imagePullPolicy }}
         resources:
 {{ toYaml $.Values.minio.metricsLogger.resources | indent 10 }}
         env:
+        - name: LOG_PROVIDER
+          value: {{ $.Values.minio.metricsLogger.LOG_PROVIDER | quote }}
+        - name: LOG_LEVEL
+          value: {{ $.Values.minio.metricsLogger.LOG_LEVEL | quote }}
         - name: REDIS_HOST
           value: {{ $.Values.minio.metricsLogger.REDIS_HOST | quote }}
         - name: REDIS_PORT
           value: {{ $.Values.minio.metricsLogger.REDIS_PORT | quote }}
-        - name: REDIS_POOL_MAX_CONNECTIONS
-          value: {{ $.Values.minio.metricsLogger.REDIS_POOL_MAX_CONNECTIONS | quote }}
-        - name: REDIS_POOL_TIMEOUT
-          value: {{ $.Values.minio.metricsLogger.REDIS_POOL_TIMEOUT | quote }}
-        - name: REDIS_KEY_PREFIX_DEPLOYMENT_LAST_ACTION
-          value: {{ $.Values.minio.metricsLogger.REDIS_KEY_PREFIX_DEPLOYMENT_LAST_ACTION | quote }}
         - name: UPDATE_GRACE_PERIOD_SECONDS
           value: {{ $.Values.minio.metricsLogger.UPDATE_GRACE_PERIOD_SECONDS | quote }}
         - name: DEPLOYMENT_API_METRICS_FLUSH_INTERVAL_SECONDS
           value: {{ $.Values.minio.metricsLogger.DEPLOYMENT_API_METRICS_FLUSH_INTERVAL_SECONDS | quote }}
+        - name: REDIS_KEY_PREFIX_DEPLOYMENT_LAST_ACTION
+          value: {{ $.Values.minio.metricsLogger.REDIS_KEY_PREFIX_DEPLOYMENT_LAST_ACTION | quote }}
         - name: REDIS_KEY_PREFIX_DEPLOYMENT_API_METRIC
           value: {{ $.Values.minio.metricsLogger.REDIS_KEY_PREFIX_DEPLOYMENT_API_METRIC | quote }}
-        {{ if $.Values.minio.metricsLogger.debug }}
-        - name: DEBUG
-          value: "yes"
-        {{ end }}
+
+        # ElasticSearch
+        # - name: ES_HOST
+        #   value: {{ $.Values.minio.metricsLogger.ES_HOST | quote }}
+        # - name: ES_PORT
+        #   value: {{ $.Values.minio.metricsLogger.ES_PORT | quote }}
+
+        # S3
+        # - name: AWS_KEY_ID
+        #   value: {{ $.Values.minio.metricsLogger.AWS_KEY_ID | quote }}
+        # - name: AWS_SECRET_KEY
+        #   value: {{ $.Values.minio.metricsLogger.AWS_SECRET_KEY | quote }}
+        # - name: S3_BUCKET_NAME
+        #   value: {{ $.Values.minio.metricsLogger.S3_BUCKET_NAME | quote }}
+        # - name: S3_REGION
+        #   value: {{ $.Values.minio.metricsLogger.S3_REGION | quote }}
```

Comment:
these need to be un-commented and wrapped in an if depending on the log provider (so the environments variables for each log provider will be passed only if that provider is selected)